### ISSUE TITLE
phpmd ignore SuppressWarnings(PHPMD.CountInLoopExpression) inside class

### ISSUE THREAD

**Issue by fernandoval** [phpmd/phpmd#676]
- PHPMD version: 2.7.0
- PHP Version: 7.2.19
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 18.04.3

### Current Behavior
Receiving "Avoid using count() function in while loops" warning message.

### Expected Behavior
No error messages.

### Steps To Reproduce:
If comment @SuppressWarnings(PHPMD.CountInLoopExpression) is in class comment block no warning message is printed.

But if the suppression instruction is in function comment block the warning message is printed.

**Comment by kylekatarnls** [comment]
Hi @fernandoval, please dump the code so we can reproduce it.

Did you check the PHPDoc comment has the valid syntax with 2 stars: `/** ... */`?

**Comment by fernandoval** [comment]
Sample of code where commet works:

```php
/**
 * @SuppressWarnings(PHPMD.CountInLoopExpression)
 */
class ArrayUtils
{
    public function dottedUnset(array &$array, string $key)
    {
        $keys = explode('.', $key);

        while (count($keys) > 1) {
            $key = array_shift($keys);

            if (!isset($array[$key]) || !is_array($array[$key])) {
                return;
            }

            $array = &$array[$key];
        }

        unset($array[array_shift($keys)]);
    }
}
```

Sample where comment does not works:

```php
class ArrayUtils
{
    /**
     * Removes the element from an array using dot notation.
     *
     * @SuppressWarnings(PHPMD.CountInLoopExpression)
     *
     * @param array  $array
     * @param string $key
     *
     * @return void
     */
    public function dottedUnset(array &$array, string $key)
    {
        $keys = explode('.', $key);

        while (count($keys) > 1) {
            $key = array_shift($keys);

            if (!isset($array[$key]) || !is_array($array[$key])) {
                return;
            }

            $array = &$array[$key];
        }

        unset($array[array_shift($keys)]);
    }
}
```

**Comment by kylekatarnls** [comment]
Bug reproduced. Fix #677 proposed.
